### PR TITLE
Add styled-system to project

### PR DESCRIPTION
**Pull request by samajammin**
## Description

- Add styled-system to project: https://styled-system.com/
- Move src/components/Theme.js → src/theme.js (since it's not actually a component)

## Related Issue
None

### CODE REVIEW

File: src/components/ButtonLink.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
 import React from "react"
 import styled from "styled-components"
+import { margin } from "styled-system"
```
Comment (samajammin) [inline src/components/ButtonLink.js:3]: FYI @ryancreatescopy - here's an example of how to add this to a component.

1) import from styled-system (e.g. `import { margin } from "styled-system"`)
2) add to a styled component (e.g. `${margin}`)
3) pass all props the component receives to the styled component (see `...props` code below)

File: src/components/ButtonLink.js [hunk 2]
```diff
@@ -42,19 +43,23 @@ const Secondary = styled(StyledButton)`
...
   return (
-    <Primary to={to} hideArrow={true} className={className}>
+    <Primary to={to} hideArrow={true} className={className} {...props}>
       {children}
     </Primary>
   )
 }
 
+ButtonLink.defaultProps = {
+  mt: `0.5rem`,
```
Comment (samajammin) [inline src/components/ButtonLink.js:62]: @ryancreatescopy curious for your thoughts on this. I added this to keep parity with the current designs (default margin-top of 0.5rem on all buttons) but I don't think this is actually needed/desired. Seems kind of strange to me to apply a margin to all buttons.

Comment (ryancreatescopy) [reply src/components/ButtonLink.js:62]: The desired margin will probably differ dependent on context. So it'd probably be best to remove a default margin IMO as you'll need to add custom margin regardless.

File: src/components/ButtonLink.js (hunk 1 above)
Comment (ryancreatescopy) [reply src/components/ButtonLink.js:3]: Ok, I think this makes sense. Looks good.

**Comment by samajammin** [comment]
@ryancreatescopy - I removed the default ButtonLink margin. This is ready for your review!

**Review by ryancreatescopy — APPROVED** [review]

**Pull request merged**